### PR TITLE
session: fix resource leak on some errors

### PR DESCRIPTION
**Pull request by nuivall**
On `connConfig` error session object is not closed so partially created resources are leaking, for instance `flusher` goroutine created by `newEventDebouncer`.

### CODE REVIEW

**Comment by nuivall** [comment]
As for testing I ran go test for this repository and also I'm running Scylla Cloud end-to-end testing with the patch now. Please let me know if more testing is needed for this patch.

File: session.go [hunk 1]
```diff
@@ -103,7 +103,7 @@ func addrsToHosts(addrs []string, defaultPort int) ([]*HostInfo, error) {
 }
 
 // NewSession wraps an existing Node.
-func NewSession(cfg ClusterConfig) (*Session, error) {
+func NewSession(cfg ClusterConfig) (_ *Session, err error) {
```
Comment (rjeczalik) [inline session.go:106]: I would just add `var err error` to L130 so no need for naming those.

https://github.com/golang/go/wiki/CodeReviewComments#named-result-parameters

Comment (nuivall) [reply session.go:106]: I think it's more robust as it's easy to `return err2` somewhere later and make the leak pop up again

Comment (mmatczuk) [reply session.go:106]: I vote for adding var err error

Comment (nuivall) [reply session.go:106]: fine

**Pull request merged**